tests/unit/src/test/scala/tests/worksheets/WorksheetSuite.scala

```diff
@@ -238,4 +247,71 @@ class WorksheetSuite extends FunSuite {
        |""".stripMargin
   )
 
+  val testTypesString =
+    """
+      |val a = 1
+      |val b: String = "A"
+      |val (x, y) = (true, 0)
+      |a + 4
+      |class C {}
+      |def f(): String = "FFF"
+      |f()
+      |
+      |def px(): String = { println("Something"); "OK" }
+      |
+      |println("print")
+      |
+      |px()
+      |
+      |{ println("P1"); List(1,2,3) }
+      |""".stripMargin
+
+  checkDecorations(
+    "types-enabled",
+    testTypesString,
+    """|
+       |<val a = 1> // 1: Int
+       |<val b: String = "A"> // "A": String
+       |<val (x, y) = (true, 0)> // x: Boolean=true, y: Int=0
+       |<a + 4> // 5: Int
+       |class C {}
+       |def f(): String = "FFF"
+       |<f()> // "FFF": String
+       |
+       |def px(): String = { println("Something"); "OK" }
+       |
+       |<println("print")> // _: Unit , out: print
+       |
+       |<px()> // "OK": String
+       |
+       |<{ println("P1"); List(1,2,3) }> // List(1, 2, 3): List[Int]
```

Comment:
It is truncated to exactly match marginWidth (added test to show how it looks)

Reply:
I played with `BlackWhite.tokenize` as it is used for value but it only splits single chunk with full type so not helpful, we could manually cut type earlier if we encounter other char than [a-zA-Z0-9] but I am unsure it would be worth it. Especially for long type name it might make sense to show as much as possible.

Reply:
Let me know if you are okay with keeping it as it is for now and improve later or do you have other ideas how to improve it?